app.py

```diff
@@ -20,27 +20,21 @@ def show_results():
     result_dict = gather_meals_by_search(search_val)
     
     return render_template('recipes.html', category="Relevant", data=result_dict)
-    # recipe_list = gather_meals()
-    # for recipe in recipe_list:
-    #     if recipe[0][0] is request.form["recipename"]:
-    #         ingredients_list = {}
-    #         for ingredient in recipe[6]:
-    #              ingredients_list = update(ingredient)
-    #         return render_template('ingredients.html', recipe_name=recipename, data=ingredients_list)
 
 # Render lucky
-# @app.route('/lucky')
-# def random():
-#     size = 0
-#     recipe_list = gather_meals()
-#     for recipe in recipe_list:
-#         size = size + 1
-#     random_recipe = recipe_list[randrang(size+1)]
-#     random_name = recipe_list[0][0]
-#     ingredients_list = {}
-#     for ingredient in random_recipe[6]:
-#         ingredient_list = update(ingredient)
-#     return render_template('lucky.html', recipe_name=random_name, data=ingredient_list)
+@app.route('/lucky', methods = ["GET", "POST"])
+def random():
+    # Dictionary of recipe names
+    recipe_dict = gather_meals()
```

Comment:
In order to make the gather_meals() call we need to add gather_meals to the library imports at the top of the page - Also we need:
`Import random` 
at the top as well.